content/swift/concepts/protocols/terms/decoder/decoder.md

```diff
@@ -0,0 +1,71 @@
+---
+Title: 'Decoder'
+Description: 'A Swift protocol that converts external data into Swift types.'
+Subjects:
+  - 'iOS'
+  - 'Mobile Development'
+Tags:
+  - 'Swift'
+  - 'iOS'
+  - 'Protocols'
+CatalogContent:
+  - 'learn-swift'
```

Comment:
```suggestion
- 'learn-swift'
- 'paths/build-ios-apps-with-swiftui'
```
Two CatlogContent slugs should be there, first for a free course and second for a pro course. You can find the CatalogContent [here](https://github.com/Codecademy/docs/blob/main/documentation/catalog-content.md).